js/script.js

```diff
@@ -0,0 +1,116 @@
+$(document).ready(function() {
+
+	var images = [];
+
+	var counter = 0;
+
+
+	
+
+	$('button').on('click', function(e) { 
+		var $input = $("#input").val();
+		// console.log(input)
+		e.preventDefault();
+		// alert("I work");
+		$.get("https://www.reddit.com/search.json?q", {
+		q: $input
+
+	}).done(function(data) {
+		//console.log(data)
+		images = data.data.children;
+		// console.log(images)
+
+		function filterByUrl(obj) {
+			if (obj.data.url) {
+				return true
+
+			}
+			else {
+				return false
+			}
+
+
+		}
+		var imagesFilter = images.filter(filterByUrl)
+		// console.log(imagesFilter)
+		var allImages = imagesFilter.map(function(obj) {
+			return obj.data.url
```

Comment:
Likewise with your filter function, you'll want to filter out posts that don't have any preview images:

```
	return obj.data.preview.images[0].source.url
```